src/actions/write/setRecord.ts

```diff
@@ -23,32 +45,32 @@ export const setRecord = async (
   }
 
   // check subdomain validity
-  const namePattern = new RegExp("^[a-zA-Z0-9_-]+$"); // include underscores and dashes
+  const namePattern = new RegExp('^[a-zA-Z0-9_-]+$'); // include underscores and dashes
```

Comment:
add AJV validations to these methods similar to what we do in testnet contract and remove these input checks